src/debugAdapter/goDebug.ts

```diff
@@ -802,7 +810,7 @@ class GoDebugSession extends DebugSession {
 			verbose('locations', locations);
 			let stackFrames = locations.map((location, i) =>
 				new StackFrame(
-					i,
+					frameId(args.threadId, i),
```

Comment:
Yes, it assumes we don't get a stack trace deeper than 256 frames.

Reply:
Delve doesnt have any such limitations AFAIK, therefore it wouldnt be safe for us to make this assumption.

Reply:
So what should we do? Any other ideas?

Reply:
If it just has to be a unique number across threads, then why not just generate a random number and then use that random number + i here?

Reply:
Just a `+` is dangerous if we generate two random numbers that are close enough together. My experience with such things is that it will happen more often than not.

Reply:
In that case, we generate a random number once. Then for every new thread, 

```
theRandomNumber += 10000
this.threads.set(threadId, theRandomNumber )
```

?

Reply:
First thread can get, lets say, 100. Second can get say 200. 100 + 10000 and 200 + 10000. Adding 10000 doesn't fix the overlap.

I don't see any real choice but to divide the key space so that each thread has a set amount of frame IDs to use. Whether that be by addition, shifting, or some other formula. That's basiclly a limitation of the debug protocl that frames are not identified by a [thread ID, frame ID] tuple like most debuggers actually really do.

We can play with the shift amount to give more space to frames in contrast to threads of course or even flip their order in the computed integer.

Reply:
That's achieves exactly the same result as shifting and `or`-ing the thread number in, which this PR does. Just with addition, and with additional complexity of having to keep track of the new ID's we give to each thread.

If we want more space for frames we can just change the shift count. For example to 12 to get 2**12 = 4096 frames (A multiple of 4 power is nicer to look at in hex).

Reply:
Makes sense, lets do the increased shift count